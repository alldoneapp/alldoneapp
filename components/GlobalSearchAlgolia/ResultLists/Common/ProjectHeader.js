import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

import ColoredCircleSmall from '../../../SidebarMenu/ProjectFolding/ProjectItem/ColoredCircleSmall'
import styles, { colors } from '../../../styles/global'
import ProjectBadge from './ProjectBadge'
import Icon from '../../../Icon'

export default function ProjectHeader({ project, amount, containerStyle }) {
    return (
        <View style={[localStyles.container, containerStyle]}>
            <View style={localStyles.titleContainer}>
                {project.color ? (
                    <ColoredCircleSmall
                        size={16}
                        color={project.color}
                        isGuide={!!project.parentTemplateId}
                        containerStyle={{ marginHorizontal: 4 }}
                        projectId={project.id}
                    />
                ) : (
                    <Icon size={24} name={'circle'} color={colors.Text03} />
                )}
                <Text style={localStyles.projectName} numberOfLines={1}>
                    {project.name}
                </Text>
                <Text style={localStyles.dot}>â€¢</Text>
                <ProjectBadge value={amount} />
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        height: 56,
        justifyContent: 'flex-end',
        paddingBottom: 6,
        borderBottomColor: colors.Grey400,
        borderBottomWidth: 1,
        marginHorizontal: 16,
    },
    titleContainer: {
        alignItems: 'center',
        justifyContent: 'flex-start',
        flexDirection: 'row',
    },
    projectName: {
        ...styles.subtitle1,
        paddingLeft: 8,
        color: '#ffffff',
    },
    dot: {
        ...styles.subtitle1,
        color: colors.Text03,
        marginHorizontal: 6,
    },
})
