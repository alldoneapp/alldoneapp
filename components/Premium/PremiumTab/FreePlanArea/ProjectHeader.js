import React from 'react'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { useSelector } from 'react-redux'

import styles, { colors } from '../../../styles/global'
import ProjectHeaderMoreButton from '../../../SettingsView/Profile/Skills/SkillsByProject/Sorting/ProjectHeaderMoreButton'
import NavigationService from '../../../../utils/NavigationService'
import store from '../../../../redux/store'
import ColoredCircleSmall from '../../../SidebarMenu/ProjectFolding/ProjectItem/ColoredCircleSmall'
import { setSelectedNavItem } from '../../../../redux/actions'
import { DV_TAB_USER_PROFILE } from '../../../../utils/TabNavigationConstants'

export default function ProjectHeader({ projectId, containerStyle, showSkillsMoreButton }) {
    const name = useSelector(state => state.loggedUserProjectsMap[projectId].name)
    const color = useSelector(state => state.loggedUserProjectsMap[projectId].color)
    const parentTemplateId = useSelector(state => state.loggedUserProjectsMap[projectId].parentTemplateId)

    const isGuide = !!parentTemplateId

    const onPressProject = () => {
        const { loggedUser, loggedUserProjectsMap } = store.getState()
        NavigationService.navigate('UserDetailedView', {
            contact: loggedUser,
            project: loggedUserProjectsMap[projectId],
        })
        store.dispatch(setSelectedNavItem(DV_TAB_USER_PROFILE))
    }

    return (
        <View style={[localStyles.container, containerStyle]}>
            <View style={localStyles.projectNameContainer}>
                <TouchableOpacity style={{ flexDirection: 'row' }} onPress={onPressProject}>
                    <ColoredCircleSmall
                        size={16}
                        color={color}
                        isGuide={isGuide}
                        containerStyle={{ margin: 4 }}
                        projectId={projectId}
                    />
                    <Text style={[styles.subtitle1, localStyles.projectName]} numberOfLines={1}>
                        {name}
                    </Text>
                </TouchableOpacity>
                {showSkillsMoreButton && <ProjectHeaderMoreButton projectId={projectId} />}
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        paddingHorizontal: 16,
    },
    projectNameContainer: {
        flex: 1,
        flexDirection: 'row',
        borderBottomColor: colors.Grey300,
        borderBottomWidth: 1,
        paddingBottom: 7,
        paddingTop: 24,
        marginTop: 15,
        marginBottom: 7,
        justifyContent: 'space-between',
    },
    projectName: {
        ...styles.subtitle1,
        marginLeft: 4,
        color: colors.Text01,
    },
})
