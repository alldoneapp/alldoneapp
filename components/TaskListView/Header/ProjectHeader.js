import React from 'react'
import { StyleSheet, View } from 'react-native'
import { useDispatch, useSelector } from 'react-redux'

import { colors } from '../../styles/global'
import { setSelectedNavItem } from '../../../redux/actions'
import NavigationService from '../../../utils/NavigationService'
import { DV_TAB_USER_WORKFLOW } from '../../../utils/TabNavigationConstants'
import TagsArea from './TagsArea'
import {
    DV_TAB_ROOT_CONTACTS,
    DV_TAB_ROOT_NOTES,
    DV_TAB_ROOT_UPDATES,
    DV_TAB_ROOT_CHATS,
} from '../../../utils/TabNavigationConstants'
import store from '../../../redux/store'
import ProjectAndUserData from './ProjectAndUserData'

export default function ProjectHeader({
    projectIndex,
    projectId,
    showWorkflowTag = false,
    badge,
    customRight,
    showAddTask,
    showAddGoal,
    setPressedShowMoreMainSection,
}) {
    const dispatch = useDispatch()

    const currentUser = useSelector(state => state.currentUser)
    const loggedUser = useSelector(state => state.loggedUser)
    const selectedTab = useSelector(state => state.selectedSidebarTab)
    const mobile = useSelector(state => state.smallScreenNavigation)
    const mobileCollapsed = useSelector(state => state.smallScreenNavSidebarCollapsed)

    const userInHeader =
        selectedTab === DV_TAB_ROOT_CONTACTS ||
        selectedTab === DV_TAB_ROOT_NOTES ||
        selectedTab === DV_TAB_ROOT_CHATS ||
        selectedTab === DV_TAB_ROOT_UPDATES
            ? loggedUser
            : currentUser

    const workflow = userInHeader.workflow ? userInHeader.workflow[projectId] : undefined

    const onClickWorkflowIndicator = () => {
        const { loggedUserProjectsMap } = store.getState()
        dispatch(setSelectedNavItem(DV_TAB_USER_WORKFLOW))
        NavigationService.navigate('UserDetailedView', {
            contact: userInHeader,
            project: loggedUserProjectsMap[projectId],
        })
    }

    const haveWorkflow = () => {
        return (
            userInHeader &&
            userInHeader.workflow &&
            userInHeader.workflow[projectId] &&
            Object.values(userInHeader.workflow[projectId]).length > 0
        )
    }

    const showWorkflow = showWorkflowTag && haveWorkflow()

    return (
        <View style={localStyles.borderContainer}>
            <View style={localStyles.container}>
                <ProjectAndUserData
                    projectIndex={projectIndex}
                    projectId={projectId}
                    badge={badge}
                    userInHeader={userInHeader}
                />
                <TagsArea
                    projectId={projectId}
                    workflow={workflow}
                    user={userInHeader}
                    mobile={mobile || mobileCollapsed}
                    onClickWorkflowIndicator={onClickWorkflowIndicator}
                    showWorkflow={showWorkflow}
                    showAddTask={showAddTask}
                    showAddGoal={showAddGoal}
                    setPressedShowMoreMainSection={setPressedShowMoreMainSection}
                />
                {customRight}
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    borderContainer: {
        borderBottomWidth: 1,
        borderBottomColor: colors.Grey300,
    },
    container: {
        flex: 1,
        height: 56,
        minHeight: 56,
        maxHeight: 56,
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingTop: 25,
        paddingBottom: 6,
    },
    subContainer: {
        maxHeight: 24,
        height: 24,
        flex: 1,
        alignItems: 'center',
        justifyContent: 'flex-start',
        flexDirection: 'row',
    },
    titleSubContainer: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    titleContainer: {
        alignItems: 'center',
        justifyContent: 'flex-start',
        flexDirection: 'row',
    },
    projectName: {
        paddingLeft: 4,
        color: colors.Text01,
    },
    userName: {
        color: colors.Text01,
    },
    compass: {
        backgroundColor: 'green',
        borderRadius: 100,
        opacity: 0,
    },
    dotSeparator: {
        width: 4,
        height: 4,
        borderRadius: 16,
        backgroundColor: colors.Text02,
        marginHorizontal: 6,
    },
    userImage: {
        height: 18,
        width: 18,
        borderRadius: 100,
        marginRight: 4,
        backgroundColor: colors.Gray400,
    },
    stepUserImage: {
        height: 16,
        width: 16,
        borderRadius: 100,
    },
    stepUserImageOutline: {
        height: 20,
        width: 20,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 100,
        backgroundColor: colors.Text03,
    },
    workflowIndicator: {
        height: 24,
        backgroundColor: colors.Grey300,
        paddingHorizontal: 4,
        borderRadius: 50,
        flexDirection: 'row',
        alignItems: 'center',
    },
    centeredRow: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    workflowIconMobile: {
        marginRight: 6,
    },
    workflowLabel: {
        color: colors.Text03,
        marginLeft: 6,
        marginRight: 8,
    },
})
