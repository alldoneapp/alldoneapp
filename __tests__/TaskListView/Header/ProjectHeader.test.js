import React from 'react'
import { Provider } from 'react-redux'
import ProjectHeader from '../../../components/TaskListView/Header/ProjectHeader'
import store from '../../../redux/store'
import { storeCurrentUser } from '../../../redux/actions'

import renderer from 'react-test-renderer'

jest.mock('firebase', () => ({ firestore: {} }));

describe('ProjectHeader component', () => {
    describe('ProjectHeader snapshot test', () => {
        it('should render correctly', () => {
            store.dispatch(storeCurrentUser({ workflow: [] }))
            const tree = renderer
                .create(
                    <Provider store={store}>
                        <ProjectHeader
                            user={{ photoURL: 'http://url.to.photo' }}
                            project={{ color: '#0055ff' }}
                        />
                    </Provider>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
