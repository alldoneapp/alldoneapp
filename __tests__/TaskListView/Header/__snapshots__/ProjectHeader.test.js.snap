// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectHeader component ProjectHeader snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#E7ECEF",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": 56,
      "justifyContent": "space-between",
      "paddingBottom": 6,
      "paddingTop": 25,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "maxWidth": false,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#0055ff",
                  "fontFamily": "alldone",
                  "fontSize": 24,
                  "opacity": 1,
                }
              }
            >
              
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 16,
                  "letterSpacing": 0.16,
                  "lineHeight": 24,
                },
                Object {
                  "color": "#04142F",
                  "paddingLeft": 4,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#4E5D78",
            "borderRadius": 16,
            "height": 4,
            "marginHorizontal": 6,
            "width": 4,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "http://url.to.photo",
              }
            }
            style={
              Object {
                "backgroundColor": "#C6CDD2",
                "borderRadius": 100,
                "height": 18,
                "marginRight": 4,
                "width": 18,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "green",
            "borderRadius": 100,
            "opacity": 0,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "alldone",
                "fontSize": 19,
                "opacity": 1,
              }
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
